e2e/single-cluster/delete_namespaces_test.go

```diff
@@ -0,0 +1,107 @@
+package singlecluster_test
+
+import (
+	"errors"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rancher/fleet/e2e/testenv"
+	"github.com/rancher/fleet/e2e/testenv/kubectl"
+)
+
+var _ = Describe("delete namespaces", func() {
+	var (
+		k               kubectl.Command
+		targetNamespace string
+		deleteNamespace bool
+	)
+
+	type TemplateData struct {
+		TargetNamespace string
+		DeleteNamespace bool
+	}
+
+	BeforeEach(func() {
+		k = env.Kubectl.Namespace(env.Namespace)
+		deleteNamespace = false
+
+		DeferCleanup(func() {
+			_, _ = k.Delete("ns", "my-custom-namespace", "--wait=false")
+		})
+	})
+
+	JustBeforeEach(func() {
+		err := testenv.ApplyTemplate(k, testenv.AssetPath("single-cluster/delete-namespace/gitrepo.yaml"),
+			TemplateData{targetNamespace, deleteNamespace})
+
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() error {
+			out, err := k.Namespace(targetNamespace).Get("configmaps")
+			if err != nil {
+				return err
+			}
+
+			if !strings.Contains(out, "app-config") {
+				return errors.New("expected configmap is not found")
+			}
+
+			return nil
+		}).ShouldNot(HaveOccurred())
+	})
+
+	When("delete namespaces is false", func() {
+		BeforeEach(func() {
+			targetNamespace = "my-custom-namespace"
+		})
+
+		It("preserves targetNamespace when GitRepo is deleted", func() {
+			out, err := k.Delete("gitrepo", "my-gitrepo", "-n", "fleet-local")
+			Expect(err).ToNot(HaveOccurred(), out)
+
+			Eventually(func() error {
```

Comment:
We could even take this one step further and replace this `Eventually` call with a `Consistently` call to ensure that the namespace still exists for a little bit (eg. a few seconds) after the `GitRepo` is deleted. This would enable us to validate that the namespace exists, not only because its deletion might not _yet_ have been caused by deleting the `GitRepo`, but rather because deleting the `GitRepo` does not affect the namespace.
See [this](https://onsi.github.io/ginkgo/#patterns-for-asynchronous-testing) for the differences between `Eventually` and `Consistently`, and more details on `Consistently` [here](https://onsi.github.io/gomega/#consistently).

Reply:
Very interesting! Nice find. I have updated it